src/odemis/gui/util/filename_pattern.py

```diff
@@ -0,0 +1,219 @@
+    fn_ptn = re.sub('}', "}}", fn_ptn)
+
+    # Detect date
+    # First check daterev, then datelng because sequence daterev + timelng might otherwise
+    # be mistaken to be number + datelng + counter
+    date_ptn = '[0-3][0-9](0[1-9]|1[0-2])20[0-9][0-9]'
+    fn_ptn = re.sub(date_ptn, "{daterev}", fn_ptn)
+
+    date_ptn = '20[0-9][0-9](0[1-9]|1[0-2])[0-3][0-9]'
+    fn_ptn = re.sub(date_ptn, "{datelng}", fn_ptn)
+
+    date_ptn = '[0-3][0-9]/(0[1-9]|1[0-2])/20[0-9][0-9]'
+    fn_ptn = re.sub(date_ptn, "{daterev_slash}", fn_ptn)
+
+    date_ptn = '20[0-9][0-9]/(0[1-9]|1[0-2])/[0-3][0-9]'
+    fn_ptn = re.sub(date_ptn, "{datelng_slash}", fn_ptn)
+
+    date_ptn = '[0-3][0-9]-(0[1-9]|1[0-2])-20[0-9][0-9]'
+    fn_ptn = re.sub(date_ptn, "{daterev_hyphen}", fn_ptn)
+
+    date_ptn = '20[0-9][0-9]-(0[1-9]|1[0-2])-[0-3][0-9]'
+    fn_ptn = re.sub(date_ptn, "{datelng_hyphen}", fn_ptn)
+
+    # Detect time h-min-s
+    time_ptn = '[0-2][0-9][0-5][0-9][0-5][0-9]'
+    fn_ptn = re.sub(time_ptn, "{timelng}", fn_ptn)
+
+    time_ptn = '[0-2][0-9]:[0-5][0-9]:[0-5][0-9]'
+    fn_ptn = re.sub(time_ptn, "{timelng_colon}", fn_ptn)
+
+    time_ptn = '[0-2][0-9]/[0-5][0-9]/[0-5][0-9]'
+    fn_ptn = re.sub(time_ptn, "{timelng_slash}", fn_ptn)
+
+    time_ptn = '[0-2][0-9]-[0-5][0-9]-[0-5][0-9]'
+    fn_ptn = re.sub(time_ptn, "{timelng_hyphen}", fn_ptn)
+
+    # Detect time h-min
+    time_ptn = '[0-2][0-9]:[0-5][0-9]'
+    fn_ptn = re.sub(time_ptn, "{timeshrt_colon}", fn_ptn)
+
+    time_ptn = '[0-2][0-9]/[0-5][0-9]'
+    fn_ptn = re.sub(time_ptn, "{timeshrt_slash}", fn_ptn)
+
+    # Don't allow separation of short time by hyphen, causes problems with some names
+
+    # If 4-digit number corresponds to time in an interval of +- 15 min
+    # around the current time, recognize it as time pattern, otherwise
+    # assume it's a counter.
+    time_ptn = '[0-2][0-9][0-5][0-9]'
+    if re.search('[0-2][0-9][0-5][0-9]', fn_ptn):
+        h = int(re.search('[0-2][0-9][0-5][0-9]', fn_ptn).group()[:2])
+        m = int(re.search('[0-2][0-9][0-5][0-9]', fn_ptn).group()[2:])
+        t_s = h * 3600 + m * 60
+        cur_t_s = int(time.strftime('%H')) * 3600 + int(time.strftime('%M')) * 60
+        if cur_t_s - 900 < t_s < cur_t_s + 900:
+            fn_ptn = re.sub(time_ptn, "{timeshrt}", fn_ptn)
+
+    # Detect count
+    cnt_ptn = '\d{1,5}'
+    # if multiple numbers are present, use last
+    if re.findall(cnt_ptn, fn_ptn[:-len(ext)]):
+        cnt = re.findall(cnt_ptn, fn_ptn[:-len(ext)])[-1]
+        # Use reverse of the string and replace first occurrence (= last occurrence),
+        # then reverse again
+        cnt_rev = re.findall(cnt_ptn, fn_ptn[-len(ext)::-1])[0]
+        fn_ptn = re.sub(cnt_rev, "}tnc{", fn_ptn[::-1], 1)[::-1]
+    else:
+        cnt = '0'
+
+    # If neither time, nor count are specified, add count, if only short time (h, min)
+    # specified and no count, add seconds to make filename unique
+    if 'cnt' not in fn_ptn:
```

Comment:
All these couples of lines, the patterns should be coming with the curly brackets too (eg `{cnt}`), otherwise you might just match fixed code like "mycnty". Ideally, it would also *not* match strings like `{{cnt}}`, but maybe we don't need to care too much (might be worthy to put this limitation in a comment).